src/hotspot/cpu/x86/x86.ad

```diff
@@ -4424,9 +4424,9 @@ instruct insert8D(vec dst, vec src, regD val, immI idx, rRegL tmp, legVec vtmp)
 
 // =======================Int Reduction==========================================
 
-instruct reductionI(rRegI dst, rRegI src1, vec src2, vec vtmp1, vec vtmp2) %{
+instruct reductionI(rRegI dst, rRegI src1, legVec src2, legVec vtmp1, legVec vtmp2) %{
   predicate(vector_element_basic_type(n->in(2)) == T_INT &&
-            vector_length(n->in(2)) < 16); // src2
+            vector_length(n->in(2)) <= 16); // src2
```

Comment:
FTR since `reduction16I` is gone, vector length check becomes redundant: `reductionI` successfully covers the whole range of vectors of int.

Reply:
Nice catch!

The same case with reductionS.

I'll fix it next time.
Thanks.